pkg/storage/vault/memory/memory.go

```diff
@@ -0,0 +1,42 @@
+package memory
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/scratchdata/scratchdata/config"
+)
+
+type MemoryVault struct {
+	destinations map[string]map[string]any
+}
+
+func NewMemoryVault(destinations []config.Destination) (*MemoryVault, error) {
+	vault := &MemoryVault{
+		destinations: make(map[string]map[string]any),
+	}
+	for _, dest := range destinations {
+		vault.destinations[strconv.Itoa(int(dest.ID))] = dest.Settings
```

Comment:
If there are multiple destinations, then this code does not work. Try it and you'll see.

Reply:
I think this was happening because I was fetching credentials using databaseId earlier, while it should be destination credential ID.

Reply:
This has been resolved now. We just need to make sure that we are using different destination IDs in config yaml.